src/main/java/org/vaadin/harry/spring/views/reportoverview/ReportOverview.java

```diff
@@ -0,0 +1,414 @@
+    private boolean isClicked_report = false;
+    @Id("table")
+    private static Grid<Report> reportTable;
+    //    @Id("infos-report")
+//    private Element detailDiv;
+//    @Id("infos-report2")
+//    private Element detailDivDiff;
+    @Id("wrapper-info")
+    private HorizontalLayout wrapperInfo;
+    private static VerticalLayout footerReport = new VerticalLayout();
+    private static HorizontalLayout footerTitle = new HorizontalLayout();
+    private static Select<String> selectPriority = new Select<>();
+    private static Select<String> status = new Select<>();
+    private static Select<String> versionSelected = new Select<>();
+    private static Select<String> selectType = new Select<>();
+    private static HorizontalLayout footerContent = new HorizontalLayout();
+    private static Text author = new Text("");
+    private static Paragraph reportDetails = new Paragraph();
+    private static Button btnUpdate = new Button("Update");
+    private static Button btnRevert = new Button("Revert");
+
+    private static AtomicReference<Set<ProjectVersion>> projectVersions = new AtomicReference(new ProjectVersion());
+    private static AtomicReference<String> projectSelected = new AtomicReference(new ArrayList<>());
+    private static ArrayList<String> listVersions = new ArrayList<>();
+    private static List<Report> reportListFilterByProjectAndVersion = new ArrayList<Report>();
+    /**
+     * Creates a new ReportOverview.
+     */
+    public ReportOverview() {
+        bugrapRepository.populateWithTestData();
+
+        this.setVisibleOverviewReport();
+        // find all projects
+        Set<Project> allProjects = this.findAllProjects();
+
+        ArrayList<String> listProjects = new ArrayList<>();
+        allProjects.stream().forEach(pr -> {
+            listProjects.add(pr.getName());
+        });
+
+        //set items to project combo box
+        projectsComboBox.setItems(listProjects);
+        projectsComboBox.setValue(listProjects.get(0));
+
+        // Titles of report table with grid
+        reportTable.addColumn(Report::getPriority).setHeader("PRIORITY");
+        reportTable.addColumn(Report::getType).setFlexGrow(0).setWidth("100px").setHeader("TYPE");
+        reportTable.addColumn(Report::getSummary).setHeader("SUMMARY");
+        reportTable.addColumn(Report::getAssigned).setFlexGrow(0).setWidth("160px").setHeader("ASSIGNED TO");
+        reportTable.addColumn(Report::getReportedTimestamp).setHeader("LAST MODIFIED").setWidth("140px");
+        reportTable.addColumn(Report::getTimestamp).setHeader("REPORTED").setWidth("140px");
+
+        this.setValueForProjectAndVersionWithoutClickEvents(projectVersions,
+                projectSelected,
+                allProjects,
+                listVersions);
+
+        // set grid report table can select muiti rows
+        reportTable.getSelectedItems();
+        reportTable.setSelectionMode(Grid.SelectionMode.MULTI);
```

Comment:
The grid shouldn't probably be multi-selectable because only one record is being processed at once.

Reply:
Ah, nevermind. There seems to also be "Mass-modification mode" feature described in the specs